blocks/recentreports/recentreports.js

```diff
@@ -215,29 +219,23 @@ function addDiscoverLink(discoverMoreDiv, block) {
   }
 }
 
-function addCardsDiv(block, blockConfig) {
-  if (blockConfig.type === 'recentreports') {
-    const carouselItems = document.createElement('div');
-    carouselItems.classList.add('carousel-items');
-    createCarouselDiv(block);
-    getDataFromAPI(getResearchAPIUrl(), 'GetResearchRecentReports', async (error, recentReportsDataArray = []) => {
-      if (error || recentReportsDataArray.length === 0) {
-        const element = block.querySelector('.carousel-wrapper');
-        handleNoResults(element);
-      } else {
-        renderRecentReportsCards(
-          recentReportsDataArray,
-          carouselItems,
-          blockConfig,
-          block?.dataset?.maxLimit,
-        );
-        observe(block, loadCarousel, carouselItems);
-      }
-    });
-  } else {
-    const section = block.closest('.section');
-    section.style.visibility = 'hidden';
-  }
+function addCardsDiv(block, title) {
+  const carouselItems = document.createElement('div');
+  carouselItems.classList.add('carousel-items');
+  getDataFromAPI(getResearchAPIUrl(), 'GetResearchRecentReports', async (error, recentReportsDataArray = []) => {
+    if (error || recentReportsDataArray.length === 0) {
+      const element = block.querySelector('.carousel-wrapper');
+      handleNoResults(element);
+    } else {
+      renderRecentReportsCards(
+        recentReportsDataArray,
+        carouselItems,
+        title,
+        block?.dataset?.maxLimit,
+      );
+      observe(block, loadCarousel, carouselItems);
```

Comment:
Do we need this `observe` to handle carousel rendering?

Reply:
@JiangLong2019 carousel is created here https://github.com/aemsites/icicidirect/pull/289/files#diff-e7ff8117e0e6827be45cbee4d72b0300fb5d7d7ac8bc8f6cbcfa26c10f7eac62R295, the dyamic data is added in the observe

Reply:
@anagarwa there are two `observe`. The first one[0] we use to call API when the block comes to viewpoint. In [0], we are calling another `observe`[1] which loads the carousel. Do you think we need the second `observe` to double check the viewpoint?

[0] - https://github.com/aemsites/icicidirect/pull/289/files#diff-e7ff8117e0e6827be45cbee4d72b0300fb5d7d7ac8bc8f6cbcfa26c10f7eac62R296
[1] - https://github.com/aemsites/icicidirect/pull/289/files#diff-e7ff8117e0e6827be45cbee4d72b0300fb5d7d7ac8bc8f6cbcfa26c10f7eac62R236

Reply:
@JiangLong2019  yes, i missed that updated the code

Reply:
Thanks